### ISSUE TITLE
[3.7] on backend, very light gray rich text with focus not readable...

### ISSUE THREAD

**Issue by GlennMartin1** [WordPress/gutenberg#9508]
**Describe the bug**
On the back-end, rich text has a special background color on focus. If the text color is "very light gray", that special background color makes the text unreadable on focus.

**To Reproduce**
1. Create a paragraph block with text
2. Choose "very light gray" text and **boldface** or _italics_
3. Choose any colored background (or no background)
4. Click the text to give focus. (it's unreadable)
5. Click off the text. (it's readable)

![focus](https://user-images.githubusercontent.com/30901407/44919010-69bbe580-ad0a-11e8-97b5-ff0851c05925.jpg)

**Comment by GlennMartin1** [comment]
No, I'm afraid the point was entirely missed. This is not about getting a bad combination of colors.

As mentioned, this is the case with **ANY** background color, as long as three conditions are met:

- very light gray text
- boldface
- the text has focus (not selected/highlighted)

Try it again. You'll see what I mean.

**Comment by GlennMartin1** [comment]
@designsimply

What you're illustrating is not what I was describing. 

Play with boldface / italics on very light gray text. That's where the issue arises.

**Comment by designsimply** [comment]
Aha, I think I see what you mean. My mistake was that, for the focus step, I had highlighted instead of just clicking on the inline boundary as your instructions mentioned. Sorry about that! 

> Click the text to give focus. (it's unreadable)

Tested and confirmed with WordPress 4.9.8 and Gutenberg 3.8.0-rc.1 that clicking on any text with an inline boundary (such as bold text) while the text color is set to `very light gray` results in text that is hard to read. ([35s](https://cloudup.com/cJhfeb49R3Z))

Screenshot showing `very light gray` text with a selected inline boundary:

![very-light-gray-text-with-selected-inline-boundary](https://user-images.githubusercontent.com/1119271/45381084-8d601500-b5c1-11e8-8c4f-6c381596b0f2.png)
<sup>Seen at http://alittletestblog.com/wp-admin/post.php?post=14870&action=edit running WordPress 4.9.8 and Gutenberg 3.8.0-rc.1 using Firefox 62.0 on macOS 10.13.6.</sup>

I am not sure if this is something easy to solve and am keeping the `Needs Design Feedback` label in order to get a designer's opinion on it.

**Comment by chrisvanpatten** [comment]
@GlennMartin1 I hope you don't mind but I edited the title to be a bit more descriptive of the problem! For reference, the original title was "[3.7] on backend, very light gray rich text with focus not readable..."

**Comment by chrisvanpatten** [comment]
One quick thought would be to use the automated contrast checker to invert the background-color if necessary, although I'm not sure if the inconsistency between light and dark text would be confusing.

Another option would be to tweak the inline boundary design—maybe just have a light border…?—to avoid using a background-color.

**Comment by tofumatt** [comment]
Ouch, this is a bit of a tough one. Changing the boundary design to a border over a background won't work (or at least will look incredibly strange and possibly buggy) for multi-line text.

I think the best option here is what @chrisvanpatten suggests:

> use the automated contrast checker to invert the background-color if necessary

It might be confusing, but it's better than illegible.

**Comment by afercia** [comment]
Not sure how the inline-boundary background can be changed via the automated contrast checker, as inline boundaries are a TinyMCE thing.

Some elements, when the caret is placed within them, get an "inline boundary". For example:
`<strong data-mce-selected="inline-boundary">this is bold</strong>`

The applied background color is `background: #e8eaeb;`. Same happens for other elements: `b, i, strong, em, del, ins, sup, sub`. A slightly different gray is used for `code`. Links get a light blue background instead.

Regardless of the _block_ background color, if the _text_ color is very light then the content of the inline boundaries is barely visible when clicked.

<img width="630" alt="screen shot 2018-10-13 at 14 46 39" src="https://user-images.githubusercontent.com/1682452/46905716-ed671580-cef7-11e8-9509-5f308b667dce.png">
<img width="628" alt="screen shot 2018-10-13 at 14 46 47" src="https://user-images.githubusercontent.com/1682452/46905717-ed671580-cef7-11e8-9108-abba81d572a4.png">

**Comment by chrisvanpatten** [comment]
@afercia I'm not 100% sure of the technical side but my thought is that the automated contrast checker could apply a class to the container which would be like "mce-boundary-invert" and we would conditionally update based on that.

Another option, which might be easier, would be to force the text color to change to black _when the inline boundary is active_. I don't think that would be too confusing… what do y'all think? Worth exploring?

**Comment by afercia** [comment]
@chrisvanpatten interesting, yes the second option would definitely be easier. If it's acceptable, from a design / UX perspective, to force a text color overriding the one set by users, I wouldn't have objections from an a11y perspective. I'd ask for design feedback first. Quickly tried it:

<img width="636" alt="screen shot 2018-10-13 at 17 41 20" src="https://user-images.githubusercontent.com/1682452/46907195-44c4b000-cf0f-11e8-96dc-9dcb6aab20b4.png">

Interestingly, inline `<code>` elements always force the text color to a dark one.
